### ISSUE TITLE
x/crypto/ssh/agent: cannot add ed25519 keys to local ssh agent process

### ISSUE THREAD

**Issue by disconsented** [golang/go#27671]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
go version go1.11 windows/amd64

### What operating system and processor architecture are you using (`go env`)?
Windows 10, AMD64

### What did you do?

If possible, provide a recipe for reproducing the error.
A complete runnable program is good.
A link on play.golang.org is best.
```go
package main

import (
	"crypto/rand"
	"crypto/rsa"
	"fmt"
	"github.com/xanzy/ssh-agent"
	"golang.org/x/crypto/ed25519"
	"golang.org/x/crypto/ssh/agent"
)

func addEd25519(a agent.Agent) error {
	_, priv, err := ed25519.GenerateKey(rand.Reader)
	if err != nil {
		return err
	}
	toadd := agent.AddedKey{
		PrivateKey: priv,
	}
	if err := a.Add(toadd); err != nil {
		return err
	}
	fmt.Println("ed25519 key added")
	return nil
}

func addRSA(a agent.Agent) error {
	priv, err := rsa.GenerateKey(rand.Reader, 2048)
	if err != nil {
		return err
	}
	toadd := agent.AddedKey{
		PrivateKey: priv,
	}
	if err := a.Add(toadd); err != nil {
		return err
	}
	fmt.Println("RSA key added")
	return nil
}

func main() {
	local, _, _ := sshagent.New()
	//soc, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
	//if err != nil {
	//	fmt.Printf("Error connecting to agent: %v\n", err)
	//	os.Exit(1)
	//}
	//local := agent.NewClient(soc)
	if err := addRSA(local); err != nil {
		fmt.Println(err)
	}
	if err := addEd25519(local); err != nil {
		fmt.Println(err)
	}
}
```

### What did you expect to see?
```
RSA key added
ed25519 key added

Process finished with exit code 0
```

### What did you see instead?
```
RSA key added
agent: unsupported key type ed25519.PrivateKey

Process finished with exit code 0
```

This may be a possible regression since #15701, however, I am using `xanzy/ssh-agent` instead of unix sockets.

**Comment by disconsented** [comment]
After editing ed25519.GenerateKey() locally and comparing it against rsa.GenerateKey() and agent.client.insertCert() I believe I know why this is happening.

The switch statement insertCert() uses is comparing references to those types rather than the types themselves for [example](https://sourcegraph.com/github.com/golang/crypto@e8f229864d71a49e5fdc4a9a134c5f85c4c33d64/-/blob/ssh/agent/client.go#L590) `case *rsa.PrivateKey:`

With this in mind, I went through and adjusted my local copy of insertCert to the below.
```
func GenerateKey(rand io.Reader) (PublicKey, *PrivateKey, error) {
	if rand == nil {
		rand = cryptorand.Reader
	}

	seed := make([]byte, SeedSize)
	if _, err := io.ReadFull(rand, seed); err != nil {
		return nil, nil, err
	}

	privateKey := NewKeyFromSeed(seed)
	publicKey := make([]byte, PublicKeySize)
	copy(publicKey, privateKey[32:])

	return publicKey, &privateKey, nil
}
```
Upon rerunning the test:
```
RSA key added
ed25519 key added

Process finished with exit code 0
```

**Comment by alex** [comment]
https://go-review.googlesource.com/c/crypto/+/219506 addresses this

**Comment by FiloSottile** [comment]
Fixed by golang/crypto@056763e48d71961566155f089ac0f02f1dda9b5a.
